src/genotyper.cc

```diff
@@ -108,15 +108,47 @@ class FormatFieldHelper : public IFormatFieldHelper {
         return ans;
     }
 
+    Status get_default_value(T* val) {
+        if (field_info.default_type == DefaultValueFiller::ZERO) {
+            *val = 0;
+        } else if (field_info.default_type == DefaultValueFiller::MISSING) {
+            switch (field_info.type) {
+                case RetainedFieldType::INT:
+                    *val = bcf_int32_missing;
+                    break;
+                case RetainedFieldType::FLOAT:
+                {
+                    float tmp = 0;
```

Comment:
Thanks for the explanation... I'm still a bit confused on why ```*val = bcf_int32_missing;``` works since there are 2 code paths which will type-check `*val` to either a `float` or an `int`, but the `rhs` value is obviously an `int`... 

Should I try to rewrite this using explicit template specialization? Or is this gnarly chunk ok for now?